### PR TITLE
Feature/reassemble

### PR DESCRIPTION
**Pull request by Bloodyhacker**

### CODE REVIEW

**Comment by makosblade** [comment]
What is this?

**Comment by makosblade** [comment]
Oh is this for getting rid of DirectoryController?

**Comment by Bloodyhacker** [comment]
The whole plan is: 1) move the current in-used api in DirectoryApi to UserDirectoryApi. 2) implement the new api in UserDirectoryController. 3) make k-js call UserDirectoryController instead of DirectoryController. 4) remove DirectoryApi and DirectoryController

**Comment by makosblade** [comment]
Sweeeeet, LGTM

File: src/main/java/com/kryptnostic/directory/v1/http/UserDirectoryApi.java [hunk 1]
```diff
@@ -122,9 +127,16 @@
      * @return
      */
     @POST( CONTROLLER + SET_ROLE_FOR_USER_PATH )
-    UUID addRoleForUser( @Path( ID ) UUID userId, @Body String role);
+    UUID addRoleForUser( @Path( ID ) UUID userId, @Body String role );
 
     @POST( CONTROLLER + SET_LOGIN )
     Response setSuccessfulFirstLogin();
 
+    @Timed
```
Comment (makosblade) [inline src/main/java/com/kryptnostic/directory/v1/http/UserDirectoryApi.java]: Actually, now that I think about it, I think this doesn't actually work unless you add something in the *Pod.java that loads this. It works if you put it on the implementing method though.

Comment (Bloodyhacker) [inline src/main/java/com/kryptnostic/directory/v1/http/UserDirectoryApi.java]: Is it only used by testing? can we just get rid of this?

Comment (makosblade) [inline src/main/java/com/kryptnostic/directory/v1/http/UserDirectoryApi.java]: Yeah, you can just get rid of it

**Comment by Bloodyhacker** [comment]
@geekbeast

File: src/main/java/com/kryptnostic/directory/v1/http/UserDirectoryApi.java [hunk 2]
```diff
@@ -34,18 +34,23 @@
...
     public static final String SET_LOGIN              = "/setlogin";
+    public static final String INITIALIZED            = "/initialized";
 
     public static final String ID_PATH                = "/{" + ID + "}";
     public static final String ID_WITH_DOT            = "/{" + ID + ":.+}";
     public static final String EMAIL_PATH             = "/" + EMAIL + "/{" + EMAIL + "}";   // +EMAIL needed to
                                                                                             // disambiguate from get
                                                                                             // user
+    public static final String REALM_PATH             = "/{" + REALM + "}";
+    public static final String REALM_PATH_WITH_DOT    = "/{" + REALM + ":.+}";
```
Comment (makosblade) [inline src/main/java/com/kryptnostic/directory/v1/http/UserDirectoryApi.java:52]: I thought we didn't need these anymore :(

Comment (Bloodyhacker) [inline src/main/java/com/kryptnostic/directory/v1/http/UserDirectoryApi.java:52]: No worries : )

**Pull request merged**